src/transforms/filter.js

```diff
@@ -16,11 +25,11 @@ exports.name = 'filter';
 exports.attributes = {
     operation: {
         valType: 'enumerated',
-        values: ['=', '<', '>'],
+        values: ['=', '<', '>', 'within', 'notwithin', 'in', 'notin'],
         dflt: '='
     },
     value: {
-        valType: 'number',
+        valType: 'any',
         dflt: 0
     },
     filtersrc: {
```

Comment:
you mean [`ids`](https://github.com/plotly/plotly.js/blob/920a333283b1ad9506e804a2e872eaa7f77c3f2e/src/traces/scatter/attributes.js#L68) ?